src/helpers/figures/charts/chart_common.ts

```diff
@@ -428,3 +437,80 @@ export function getDefinedAxis(definition: ChartWithAxisDefinition): {
   useLeftAxis ||= !useRightAxis;
   return { useLeftAxis, useRightAxis };
 }
+
+export function getTrendDatasetForBarChart(config: TrendConfiguration, dataset: any) {
+  const values: number[] = [];
+  const labels: number[] = [];
+  for (let i = 0; i < dataset.data.length; i++) {
+    values.push(dataset.data[i]);
+    labels.push(i + 1);
+  }
+
+  const datasetLength = dataset.data.length;
+  const newLabels: number[] = [];
+  for (let i = 0.5; i < datasetLength + 0.5; i += 0.1) {
+    newLabels.push(i);
+  }
+
+  const newValues = interpolateData(config, values, labels, newLabels);
+  if (!newValues.length) {
+    return;
+  }
+  return getFullTrendingLineDataSet(dataset, config, newValues);
+}
+
+export function getFullTrendingLineDataSet(
+  dataset: any,
+  config: TrendConfiguration,
+  data: number[]
+) {
+  return {
+    ...dataset,
+    type: "line",
+    xAxisID: "x1",
+    label: _t("Trend line for %s", dataset.label ?? ""),
+    data,
+    order: -1,
+    pointRadius: 0,
+    backgroundColor: config.color ?? lightenColor(dataset.backgroundColor, 0.5),
+    borderColor: config.color ?? lightenColor(dataset.backgroundColor, 0.5),
+    borderDash: [5, 5],
+  };
+}
+
+export function interpolateData(
+  config: TrendConfiguration,
+  values: number[],
+  labels: number[],
+  newLabels: number[]
+): number[] {
+  switch (config.type) {
+    case "polynomial": {
+      const order = config.order ?? 2;
+      const coeffs = polynomialRegression(values, labels, order, true).flat();
+      return newLabels.map((v) => evaluatePolynomial(coeffs, v, order));
+    }
+    case "linear": {
```

Comment:
Yes, it can be, but the computation method used here for the linear model is really faster than the one for the polynomial model. So, as the method were already there, I chose to separate both (just like GSheet and Excel does, they two have polynomial and linear models)

Reply:
It makes sense in all cases to split both in the UI (people don't necessarily know what is a polynome and even less that a 1st order polynome is linear), but it could be only one thing in the background.
Shouldn't "polynomial order=1" also go through the linear case then if it's faster ?